lib/rubocop/cop/rspec/pending_block_inside_example.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for passing a block by `pending` or `skip` within examples.
+      #
+      # @example
+      #   # bad
+      #   it 'does something' do
+      #     pending 'not yet implemented' do
+      #       do_something
+      #     end
+      #   end
+      #
+      #   # good
+      #   it 'does something' do
+      #     pending 'not yet implemented'
+      #     do_something
+      #   end
+      #
+      #   # bad
+      #   it 'does something' do
+      #     skip 'not yet implemented' do
+      #       do_something
+      #     end
+      #   end
+      #
+      #   # good
+      #   it 'does something' do
+      #     skip 'not yet implemented'
+      #     do_something
+      #   end
+      #
+      #   # good - when outside example
+      #   pending 'not yet implemented' do
+      #   end
+      #
+      class PendingBlockInsideExample < Base
+        MSG = "Don't pass a block to `pending` or `skip` inside examples."
+        RESTRICT_ON_SEND = %i[pending skip].freeze
+
+        def on_send(node)
+          return unless inside_example?(node)
+
+          add_offense(node.block_node)
+        end
+
+        private
+
+        def inside_example?(node)
+          Examples.all(first_left_sibling(node)&.method_name)
+        end
+
+        def first_left_sibling(node)
+          return unless (block_node = node.block_node)
+
+          block_node.left_siblings.first
+        end
```

Comment:
```suggestion
        def on_block(node)
          return unless inside_example?(node)

          add_offense(node) if %i[pending skip].include?(node.method_name)
        end

        private

        def inside_example?(node)
          node.each_ancestor(:block).any? { |ancestor| example?(ancestor) }
        end
```

Suggesting a bigger change to:
 - reduce some complexity (I couldn't quickly grasp the left sibling method)
 - only check block nodes

A viable alternative is to keep the send check restricted to `skip`/`pending` names, and use `node.parent.each_ancestor` in `inside_example?`, should also work. Leaving the decision up to you.

Reply:
I think your suggestion is certainly very good! 
Let's go with your suggestion. Thank you very much.